python/tests/test_highlevel.py

```diff
@@ -1949,6 +1949,27 @@ def test_simplify_migrations_fails(self):
         with pytest.raises(_tskit.LibraryError):
             ts.simplify()
 
```

Comment:
perhaps these should go [here](https://github.com/tskit-dev/tskit/blob/main/python/tests/test_tables.py#L4552)? And could call [`verify_subset`](https://github.com/tskit-dev/tskit/blob/main/python/tests/test_tables.py#L4614) there?

Reply:
Hmm, I could do, but I thought test_tables.py was specifically for testing TableCollection functions. The new functionality only applies to the ts.subset function (should it? Shouldn't the TableCollection function mirror the TreeSequence one exactly?). That was my rationale for dumping the tests in `test_highlevel.py`.

Reply:
Hm, good point. Um, I guess I think we should call `sort` in TableCollection.subset, instead of just here?

Reply:
Yep, that would probably be my preference.

There may be other functions which have a table and a tree sequence version, but where extra processing is required to make a valid tree sequence. It could be worth establishing a precedence for functions like this, if there is a use-case for not doing the extra processing in the tables version. We could have a parameter `make_valid_ts` that any such table function has, which is always set to `True` if called via the ts version.

It may not be that this isn't required for this particular function, so we could punt any decision down the line.

Reply:
I thought about this, and I don't think we have a use case for doing `subset` followed by other processing steps that don't require sorted tables before `sort`ing (if we did, then it could be useful for efficiency to keep it out). So, let's not add the extra parameter.